screen_brightness_control/__init__.py

```diff
@@ -454,7 +531,8 @@ def fade_brightness(
         else:
             # As `value` doesn't hit `finish` in loop, we explicitly set brightness to `finish`.
             # This also avoids an unnecessary sleep in the last iteration.
-            self.set_brightness(finish, force=force)
+            if not stoppable or threading.current_thread() == self._fade_thread:
+                self.set_brightness(finish, force=force)
 
         return self.get_brightness()
```

Comment:
Both `fade_brightness` and `fade_brightness_thread` return the new brightness, resulting in two calls to `get_brightness`.

Since returning the new brightness is deprecated for `fade_brightness` anyway, this one should probably return `None`